Lib/zipfile.py

```diff
@@ -1580,10 +1580,10 @@ def extractall(self, path=None, members=None, pwd=None):
         """Extract all members from the archive to the current working
            directory. `path' specifies a different directory to extract to.
            `members' is optional and must be a subset of the list returned
-           by namelist().
+           by infolist().
```

Comment:
I think more doc is needed (here and in the ReST documentation) to make things clear.  The code change in this PR does not change the function interface: people could pass a list of names or a list of ZipInfos before (thanks to extractall calling _extract_member which supports both), and they’re still able to do that after the change.

Maybe the doc could be explicit about these two options?  Instead of naming directly namelist or infolist (and letting people think that members must be strings or must be ZipInfos), something like «members is optional and must contain strings or ZipInfo objects corresponding to valid entries (i.e. a subset of the list returned by namelist or infolist)».

Reply:
Thanks. I juggled the idea but everything I attempted just got long winded and more confusing. Just saying `infolist` or `namelist` is both correct and really brief. Too brief?